File: code_samples/lua/error_handling_custom_exceptions.lua
-- Error Handling: Custom Exceptions
-- AI/ML Training Sample

CustomExceptions = {}
CustomExceptions.__ index = CustomExceptions

function CustomExceptions:new()
    local obj = {
        data = ""
    }
    setmetatable(obj, self)
    return obj
end

function CustomExceptions:process(input)
    self.data = input
end

function CustomExceptions:getData()
    return self.data
end

function CustomExceptions:validate()
    return #self.data > 0
end

-- Example usage
local instance = CustomExceptions:new()
instance:process("example")
print("Data: " .. instance:getData())
print("Valid: " .. tostring(instance:validate()))

File: code_samples/go/error_handling_custom_exceptions.go
package main

import (
    "fmt"
)

// Error Handling: Custom Exceptions
// AI/ML Training Sample

type CustomExceptions struct {
    Data string
}

func NewCustomExceptions() *CustomExceptions {
    return &CustomExceptions{
        Data: "",
    }
}

func (s *CustomExceptions) Process(input string) {
    s.Data = input
}

func (s *CustomExceptions) Validate() bool {
    return len(s.Data) > 0
}

func (s *CustomExceptions) GetData() string {
    return s.Data
}

func main() {
    instance := NewCustomExceptions()
    instance.Process("example")
    fmt.Printf("Data: %s\n", instance.GetData())
    fmt.Printf("Valid: %v\n", instance.Validate())
}
